### PR TITLE
[DRAFT] Adding convention based operations shared libraries (1/X)

### PR DESCRIPTION
**Pull request by jamdavi**
Add the base libraries and implementations for the Convention Based Operations, otherwise known as PnP.

This will be one of many PRs to the `jamdavi/pnpDev` branch which will eventually be merged into the `previews/pnpApi` branch. Which will of course make it to `preview` and then `master`.

### CODE REVIEW

File: deps/src/main/java/com/microsoft/azure/sdk/iot/deps/convention/ConventionConstants.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.microsoft.azure.sdk.iot.deps.convention;
+
+public final class ConventionConstants
+{
+    /// <summary>
+    /// Separator for a component-level command name.
+    /// </summary>
+    public static final char ComponentLevelCommandSeparator = '*';
```
Comment (drwill-ms) [inline deps/src/main/java/com/microsoft/azure/sdk/iot/deps/convention/ConventionConstants.java:8]: Why not const?

File: deps/src/main/java/com/microsoft/azure/sdk/iot/deps/convention/ConventionConstants.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package com.microsoft.azure.sdk.iot.deps.convention;
+
+public final class ConventionConstants
```
Comment (drwill-ms) [inline deps/src/main/java/com/microsoft/azure/sdk/iot/deps/convention/ConventionConstants.java:3]: Why not static?

File: deps/src/main/java/com/microsoft/azure/sdk/iot/deps/convention/DefaultPayloadConvention.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.microsoft.azure.sdk.iot.deps.convention;
+
+import lombok.Getter;
+
+/// <summary>
```
Comment (drwill-ms) [inline deps/src/main/java/com/microsoft/azure/sdk/iot/deps/convention/DefaultPayloadConvention.java:5]: Does Java support the same /// doc comments as C#??

**Comment by jamdavi** [comment]
Closing draft to create a better set of PRs

**Pull request closed without merging**